java/core/src/main/java/com/microsoft/bond/SetBondType.java

```diff
@@ -148,6 +154,21 @@ protected final void serializeValue(SerializationContext context, Set<TElement>
     }
 
     @Override
+    protected final Set<TElement> deserializeValue(UntaggedDeserializationContext context) throws IOException {
+        final int count = context.reader.readContainerBegin();
+        Set<TElement> value = newDefaultValue(count);
```

Comment:
Attempting to tune this is well outside the scope of this PR. Changed back to default constructors.